presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSpiller.java

```diff
@@ -51,14 +52,20 @@
     private final Path targetDirectory;
     private final Closer closer = Closer.create();
     private final BlockEncodingSerde blockEncodingSerde;
+    private final AtomicLong spilledDataSize;
 
     private int spillsCount;
     private final ListeningExecutorService executor;
 
-    public BinaryFileSpiller(BlockEncodingSerde blockEncodingSerde, ListeningExecutorService executor, Path spillPath)
+    public BinaryFileSpiller(
+            BlockEncodingSerde blockEncodingSerde,
+            ListeningExecutorService executor,
+            Path spillPath,
+            AtomicLong spilledDataSize)
```

Comment:
Can you elaborate? What's your idea? Reversing the interface would require that Factory keeps track of all created spillers or I would have to add some new proxy interface that would basically be an AtomicLong wrapper.

I was thinking about introducing `SpillingStats` object, but as long it is single long it seemed like an overkill.